packages/dd-trace/src/appsec/blocking.js

```diff
@@ -0,0 +1,40 @@
+'use strict'
+
+const fs = require('fs')
+let templateHtml, templateJson
+function block (config, req, res, topSpan, abortController) {
+  let type
+  let body
+
+  // parse the Accept header, ex: Accept: text/html, application/xhtml+xml, application/xml;q=0.9, */*;q=0.8
+  const accept = req.headers.accept && req.headers.accept.split(',').map((str) => str.split(';', 1)[0].trim())
+
+  if (accept && accept.includes('text/html') && !accept.includes('application/json')) {
+    type = 'text/html'
+    if (!templateHtml) {
+      templateHtml = fs.readFileSync(config.appsec.blockedTemplateHtml)
```

Comment:
You are right, we have to avoid this kind of Sync methods in requests, I'm going to change it to load templates at startup of the application, and not in the first blocked request.

Reply:
Changed